pandas/core/groupby/grouper.py

```diff
@@ -311,7 +311,7 @@ def _get_grouper(self, obj, validate: bool = True):
         )
         return self.binner, self.grouper, self.obj
 
-    def _set_grouper(self, obj: FrameOrSeries, sort: bool = False):
+    def _set_grouper(self, obj: FrameOrSeries, sort: bool = False, group_indices: Dict = None):
```

Comment:
We have to hand the group_indices over, otherwise we can not know with which index positions our series corresponds. The connection would be lost otherwise.